### PR TITLE
Tdi 40568 snowflake action on table impl

### PR DESCRIPTION
**Pull request by ypiel-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-40568

There were no action on table for tSnowflakeOutput nor tSnowflakeBulkExec, so can't create table on the fly.

**What is the new behavior?**
All action on table are available.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/75/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/75/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/76/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/76/)

File: components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeTableProperties.java [hunk 1]
```diff
@@ -111,4 +118,13 @@ public ValidationResult afterTableName() throws Exception {
     public SnowflakeConnectionProperties getConnectionProperties() {
         return connection;
     }
+
+
+    public void setTableaActionProvider(TableActionProvider tableaActionProvider){
```
Comment (undx) [inline components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeTableProperties.java]: Typo?

File: components/components-snowflake/components-snowflake-runtime/src/main/java/org/talend/components/snowflake/runtime/SnowflakeWriter.java [hunk 2]
```diff
@@ -111,17 +122,77 @@ public void open(String uId) throws IOException {
...
+            if (null != f.getProp(SchemaConstants.TALEND_COLUMN_IS_KEY)) {
+                keyStr.add(fName);
+            }
+        }
+
+        row = new Object[columnsStr.size()];
+
+        loader.setProperty(LoaderProperty.columns, columnsStr);
+
+        if (sprops.outputAction.getValue() == UPSERT) {
```
Comment (undx) [inline components/components-snowflake/components-snowflake-runtime/src/main/java/org/talend/components/snowflake/runtime/SnowflakeWriter.java]: Could use a flipped `equals`...

File: components/components-snowflake/components-snowflake-runtime/src/main/java/org/talend/components/snowflake/runtime/SnowflakeWriter.java [hunk 3]
```diff
@@ -257,6 +341,11 @@ protected TSnowflakeOutputProperties getProps() {
 
         prop.put(LoaderProperty.remoteStage, "~");
 
+        TableActionEnum selectedTableAction = outputProperties.tableAction.getValue();
+        if (selectedTableAction == TableActionEnum.TRUNCATE) {
```
Comment (undx) [inline components/components-snowflake/components-snowflake-runtime/src/main/java/org/talend/components/snowflake/runtime/SnowflakeWriter.java]: same (eq)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/77/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/77/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/78/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/78/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/79/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/79/)

**Pull request closed without merging**